docs/protocol/on-chain/data-structures.mdx

```diff
@@ -0,0 +1,208 @@
+---
+sidebar_position: 2
+title: "Data Structures"
+---
+
+import Tooltip from '@site/src/components/Tooltip';
+
+# Data Structures
+
+## Channel
+
+Represents the configuration of a <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">state channel</Tooltip>.
+
+```solidity
+struct Channel {
+    address[] participants;  // List of participants in the channel
+    address adjudicator;     // Contract that validates state transitions
+    uint64 challenge;        // Duration in seconds for dispute resolution
+    uint64 nonce;           // Unique identifier for the channel
+}
+```
+
+**Fields**:
+
+- `participants`: An ordered array of <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participant</Tooltip> addresses. Index 0 is typically the <Tooltip content="Most often a light client willing to fund a ledger account with a clearnode.">Creator</Tooltip>, index 1 is the <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users.">clearnode</Tooltip>.
+- `adjudicator`: Address of the <Tooltip content="A smart contract that validates state transitions according to application-specific rules.">adjudicator</Tooltip> contract responsible for validating state transitions.
+- `challenge`: Challenge period duration in seconds. Determines how long a party has to respond to a challenge.
+- `nonce`: A unique number that, combined with other fields, creates a unique <Tooltip content="A unique identifier for a channel, computed as the hash of the channel configuration.">channel identifier</Tooltip>.
+
+:::info Participant Addresses
+The caller (EOA or contract) is usually different from the <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participant</Tooltip> address. Participants are derived addresses generated by the owner, similar to <Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps.">session keys</Tooltip>, to enable high-frequency state signing without exposing the main wallet.
```

Comment:
Actually, this will become obsolete in v0.5.0 release. First participant will always be user's wallet address.
Thus, I suggest to describe the "Participant vs Caller" difference

```suggestion
:::info Participant versus Caller address
The first <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participant</Tooltip> address  is usually different from the caller (EOA or contract), thus enabling channel operation delegation. This can be fruitful as users can fund channels for other ones.
```